backend/app/Http/Controllers/UsuarioController.php

```diff
@@ -103,5 +103,38 @@ public function destroy(Request $request, string $id){
         }
     }
 
+    public function checkLogin(Request $request)
+    {
+        $request->validate([
+            'cpf' => ['required'],
+            'senha' => ['required'],
+        ]);
+        
+        $usuarios = Usuario::all();
+        
+        $validoCpf = false;
+        $validaSenha = false;
+
+        $count = 0;
+        $countCpf = -1;
+        $countSenha = -1;
+        
+        foreach ($usuarios as $usuario) {
```

Comment:
Fazendo a correção do comentário em cima você vai tirar esse for e reduzir drasticamente o numero de linhas.